app/javascript/components/bathroom.jsx

```diff
@@ -20,34 +18,36 @@ const Column = styled.div`
   margin-top: 100px;
 `;
 
-export const Bathroom = ({ cityName }) => (
-  <Query query={getBathroomCode} variables={{ cityName }}>
-    {({ loading, error, data }) => {
-      if (loading) {
-        return <LoadingMessage />;
-      }
-
-      if (error) {
-        return <ErrorMessage />;
-      }
-
-      const { bathroomCode } = data.location;
-      if (!bathroomCode) {
-        return null;
-      }
+export const Bathroom = ({ loading, error, location }) => {
+  if (loading) {
+    return <LoadingMessage />;
+  }
+  if (error) {
+    return <ErrorMessage />;
+  }
 
-      return (
-        <Column>
-          <WhiteSubTitle>BATHROOM CODE</WhiteSubTitle>
-          <WhiteTitleLarge>{bathroomCode}</WhiteTitleLarge>
-        </Column>
-      );
-    }}
-  </Query>
-);
+  const { bathroomCode } = location;
+  if (!bathroomCode) {
+    return null;
+  }
+  return (
+    <Column>
+      <WhiteSubTitle>BATHROOM CODE</WhiteSubTitle>
+      <WhiteTitleLarge>{bathroomCode}</WhiteTitleLarge>
+    </Column>
+  )
+};
 
 Bathroom.propTypes = {
-  cityName: PropTypes.string.isRequired,
+  location: PropTypes.shape({
+    bathroomCode: PropTypes.string,
+  }),
+  loading: PropTypes.bool,
+  error: PropTypes.bool,
 };
 
-export default Bathroom;
+Bathroom.fragments = {
+  location: getBathroomCode
+}
```

Comment:
Might be worth while to investigate using the `relay` client here instead, since it seems like we're emulating that pattern  here. It might provide some nice features we can use, and help organize these fragment dependencies better. 

https://relay.dev/docs/guided-tour/rendering/fragments/

cc @cbarber

Reply:
@chewnoill 
* What are the other nice features it offers over apollo?
* How does it organize the fragment dependencies better?
* Does it support subscriptions in addition to queries?
* What's the difficulty of switching out apollo for relay in our entire frontend?

Reply:
* [loading states with suspense](https://relay.dev/docs/guided-tour/rendering/loading-states/) seems pretty cool
* A main query component introspects named fragments, from its child component tree. Pretty interesting actually, basically the same concept as withFragments I think, but with extra hooks into the VDOM.
* Yes
* This is what I want investigated.

Reply:
@tmsavage don't worry about this discussion. Its out of scope of your work.